### PR TITLE
[Keyboard] Add keyboard Getta25

### PR DESCRIPTION
**Pull request by Salicylic-acid3**
Add Getta25 keyboard.

A 25 keys Tenkey.
Salicylic-acid3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/getta25/info.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                 {"label":"F2", "x":4.25, "y":3.25},
+                 {"label":"1", "x":0, "y":4.25},
+                 {"label":"2", "x":1, "y":4.25},
+                 {"label":"3", "x":2, "y":4.25},
+                 {"label":"Enter", "x":3, "y":4.25, "h":2},
+                 {"label":"DEL", "x":4.25, "y":4.25},
+                 {"label":"0", "x":0, "y":5.25, "w":2},
+                 {"label":".", "x":2, "y":5.25},
+                 {"label":"BSPC", "x":4.25, "y":5.25}
+               ]
```
Comment (noroadsleft) [inline keyboards/getta25/info.json]:
```suggestion
```

File: keyboards/getta25/readme.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# getta25
+
+![getta25](https://cdn-ak.f.st-hatena.com/images/fotolife/S/Salicylic_acid3/20191127/20191127005608.png)
+
+This is 25 keys tenkeypad. 
+
+Keyboard Maintainer: [Salicylic_acid3](https://github.com/Salicylic-acid3)
```
Comment (noroadsleft) [inline keyboards/getta25/readme.md]:
```suggestion
* Keyboard Maintainer: [Salicylic_acid3](https://github.com/Salicylic-acid3)
```

File: keyboards/getta25/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Hardware Supported: Naked64SF PCB, Pro Micro
```
Comment (noroadsleft) [inline keyboards/getta25/readme.md]:
```suggestion
* Hardware Supported: Getta25 PCB, Pro Micro
```

File: keyboards/getta25/readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+Hardware Availability: [PCB & Case Data](https://github.com/Salicylic-acid3/PCB_Data), [Booth Shop](https://salicylic-acid3.booth.pm/items/1700006)
```
Comment (noroadsleft) [inline keyboards/getta25/readme.md]:
```suggestion
* Hardware Availability: [PCB & Case Data](https://github.com/Salicylic-acid3/PCB_Data), [Booth Shop](https://salicylic-acid3.booth.pm/items/1700006)
```

File: keyboards/getta25/info.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                 {"label":"=", "x":2, "y":0},
+                 {"label":"DEL", "x":3, "y":0},
+                 {"label":"Num Lock", "x":0, "y":1.25},
+                 {"label":"/", "x":1, "y":1.25},
+                 {"label":"*", "x":2, "y":1.25},
+                 {"label":"-", "x":3, "y":1.25},
+                 {"label":"7", "x":0, "y":2.25},
+                 {"label":"8", "x":1, "y":2.25},
+                 {"label":"9", "x":2, "y":2.25},
+                 {"label":"+", "x":3, "y":2.25, "h":2},
```
Comment (noroadsleft) [inline keyboards/getta25/info.json]: Your layout macro in `getta25.h` is in a different order. We need to move some of these objects around.

```suggestion
```

File: keyboards/getta25/info.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                 {"label":"ESC", "x":4.25, "y":2.25},
+                 {"label":"4", "x":0, "y":3.25},
+                 {"label":"5", "x":1, "y":3.25},
+                 {"label":"6", "x":2, "y":3.25},
+                 {"label":"F2", "x":4.25, "y":3.25},
```
Comment (noroadsleft) [inline keyboards/getta25/info.json]:
```suggestion
                 {"label":"+", "x":3, "y":2.25, "h":2},
                 {"label":"F2", "x":4.25, "y":3.25},
```

File: keyboards/getta25/info.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                 {"label":"1", "x":0, "y":4.25},
+                 {"label":"2", "x":1, "y":4.25},
+                 {"label":"3", "x":2, "y":4.25},
+                 {"label":"Enter", "x":3, "y":4.25, "h":2},
```
Comment (noroadsleft) [inline keyboards/getta25/info.json]:
```suggestion
```

File: keyboards/getta25/info.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                 {"label":"DEL", "x":4.25, "y":4.25},
+                 {"label":"0", "x":0, "y":5.25, "w":2},
+                 {"label":".", "x":2, "y":5.25},
+                 {"label":"BSPC", "x":4.25, "y":5.25}
```
Comment (noroadsleft) [inline keyboards/getta25/info.json]:
```suggestion
                 {"label":"Enter", "x":3, "y":4.25, "h":2},
                 {"label":"BSPC", "x":4.25, "y":5.25}
```

File: keyboards/getta25/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,68 @@
+#include QMK_KEYBOARD_H
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+extern uint8_t is_master;
```
Comment (drashna) [inline keyboards/getta25/keymaps/default/keymap.c]: This block isn't needed. Especially the "is_master", as this is for split keyboards.

File: keyboards/getta25/keymaps/oled/keymap.c [hunk 10]
```diff
@@ -0,0 +1,228 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+extern uint8_t is_master;
```
Comment (drashna) [inline keyboards/getta25/keymaps/oled/keymap.c]: This block isn't needed.

File: keyboards/getta25/keymaps/oled/glcdfont.c [hunk 11]
```diff
@@ -0,0 +1,232 @@
+// 'loveLain', 128x32px
+
+#include <avr/io.h>
```
Comment (fauxpark) [inline keyboards/getta25/keymaps/oled/glcdfont.c]:
```suggestion
#include "progmem.h"
```

File: keyboards/getta25/keymaps/oled/glcdfont.c [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+#include <avr/pgmspace.h>
```
Comment (fauxpark) [inline keyboards/getta25/keymaps/oled/glcdfont.c]:
```suggestion
```

File: keyboards/getta25/keymaps/default/keymap.c (hunk 9 above)
Comment (Salicylic-acid3) [reply keyboards/getta25/keymaps/default/keymap.c]: I deleted that code block.

File: keyboards/getta25/keymaps/oled/keymap.c (hunk 10 above)
Comment (Salicylic-acid3) [reply keyboards/getta25/keymaps/oled/keymap.c]: I deleted that code block.

File: keyboards/getta25/keymaps/oled/keymap.c [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+void render_layer_messages(void) {
+    oled_write_P(PSTR("GETtA 25       For  Your Good Job. "), false);
+}
+
+void render_status(void) {
+    /* Show Keyboard Layout  */
+    render_layer_state();
+    render_keylock_status(host_keyboard_leds());
```
Comment (fauxpark) [inline keyboards/getta25/keymaps/oled/keymap.c]:
```suggestion
    render_keylock_status(host_keyboard_led_state());
```

File: keyboards/getta25/keymaps/oled/keymap.c [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+void render_layer_state(void) {
+    oled_write_P(PSTR("     "), false);
+    oled_write_P(PSTR("LAYER"), false);
+    oled_write_P(PSTR("Arrow"), layer_state_is(_ARROW));
+    oled_write_P(PSTR("Macro"), layer_state_is(_MACRO));
+    oled_write_P(PSTR("Adjus"), layer_state_is(_ADJUST));
+    oled_write_P(PSTR("     "), false);
+}
+
+void render_keylock_status(uint8_t led_usb_state) {
```
Comment (fauxpark) [inline keyboards/getta25/keymaps/oled/keymap.c]:
```suggestion
void render_keylock_status(led_t led_state) {
```

File: keyboards/getta25/keymaps/oled/keymap.c [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+    oled_write_P(PSTR("NumL "), led_usb_state & (1 << USB_LED_NUM_LOCK));
```
Comment (fauxpark) [inline keyboards/getta25/keymaps/oled/keymap.c]:
```suggestion
    oled_write_P(PSTR("NumL "), led_state.num_lock);
```

**Comment by Salicylic-acid3** [comment]
Do you still have any suggestions?

**Comment by noroadsleft** [comment]
Nope.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**